xapian-applications/omega/index_file.cc

```diff
@@ -532,11 +533,14 @@ index_add_document(const string & urlterm, time_t last_altered,
 void
 index_mimetype(const string & file, const string & urlterm, const string & url,
 	       const string & ext,
-	       const string &mimetype, DirectoryIterator &d,
-	       Xapian::Document & newdocument,
+	       const string & mime_type,
+	       DirectoryIterator & d,
+	       const string & path_term,
 	       string record)
 {
     string context(file, root.size(), string::npos);
+    string mimetype = mime_type;
+    string pathterm = path_term;
```

Comment:
You can handle this more simply by making the parameter type `string` (like the existing `record` parameter).

That wouldn't be suitable for a public API as we'd need to keep the function signature compatible, and your approach here would be necessary.  But here we can just recompile the code which calls this.

Reply:
Yes, you are right. I don't know why I do it that way. I will make that change.